sqlitecluster/SQLiteNode.cpp

```diff
@@ -1716,7 +1716,14 @@ void SQLiteNode::_onMESSAGE(SQLitePeer* peer, const SData& message) {
                     uint64_t threadAttemptStartTimestamp = STimeNow();
                     thread(&SQLiteNode::_replicate, this, peer, message, _dbPool->getIndex(false), threadAttemptStartTimestamp).detach();
                 } catch (const system_error& e) {
-                    SWARN("Caught system_error starting _replicate thread with " << _replicationThreadCount.load() << " threads. e.what()=" << e.what());
+                    // If the server is strugling and falling behind on replication, we might have too many threads
+                    // causing a resource exhaustion. If that happens, all the transations that are already threaded
+                    // and waiting for the transaction that failed will be stuck in an infinite loop. To prevent that
+                    // we're changing the state to SEARCHING and sending the cancelAfter property to drop all threads
+                    // that depend on the transaction that failed to be threaded.
+                    uint64_t cancelAfter = message.calcU64("NewCount") - 1;
```

Comment:
Might want to check NewCount against `0` first, but this is an edge case that I don't think we can actually trigger.